client/src/main/java/org/threadly/litesockets/client/http/HTTPStreamClient.java

```diff
@@ -0,0 +1,259 @@
+    client.setConnectionTimeout(timeout);
+    client.setSSLEngine(SSLUtils.OPEN_SSL_CTX.createSSLEngine(host, port));
+    client.startSSL();
+    client.setReader(classReader);
+    client.addCloseListener(classCloser);
+    httpProcessor = new HTTPResponseProcessor();
+    httpProcessor.addHTTPRequestCallback(requestCB);
+  }
+
+  /**
+   * <p>Tell the client to write an HTTPRequest to the server.  This can technically be done
+   * whenever you want to but obviously use only when you know you can sent a request, right after
+   * opening a connection for example.</p>
+   * 
+   * @param request the request to send.  This is generally done with HTTPRequestBuilders buildHeadersOnly() method.  
+   * You can send a full request if you expect the response (not the sending) to be a stream.
+   * @return returns a ListenableFuture with the HTTPResponse in it.  This might not callback right away, 
+   * especially if you used buildHeadersOnly() as you still need to stream in your data till its complete.  
+   * Once complete this should be called.
+   */
+  public ListenableFuture<HTTPResponse> writeRequest(HTTPRequest request) {
+    if(slfResponse != null && !slfResponse.isDone()) {
+      slfResponse.setFailure(new Exception("New request came in!"));
+    }
+    currentHttpRequest = request;
+    slfResponse = new SettableListenableFuture<HTTPResponse>();
+    localMbb.discard(localMbb.remaining());
+    tbb.discard(tbb.remaining());
+    client.write(request.getByteBuffer());
+    return slfResponse;
+  }
+  
+  @Override
+  public ListenableFuture<?> write(ByteBuffer bb) {
+    if(currentHttpRequest != null && currentHttpRequest.getHTTPHeaders().isChunked()) {
+      return client.write(HTTPUtils.wrapInChunk(bb));
+    } else {
+      return client.write(bb);
+    }
+  }
+  
+  @Override
+  public MergedByteBuffers getRead() {
+    return localMbb.duplicateAndClean();
+  }
+
+  @Override
+  public Executor getClientsThreadExecutor() {
+    //We use the clients executer to keep things single threaded between both clients
+    return client.getClientsThreadExecutor();
+  }
+
+  @Override
+  public boolean canWrite() {
+    return client.canWrite();
+  }
+
+  @Override
+  public boolean hasConnectionTimedOut() {
+    return client.hasConnectionTimedOut();
+  }
+
+  @Override
+  public boolean setSocketOption(SocketOption so, int value) {
+    return client.setSocketOption(so, value);
+  }
+
+  @Override
+  public ListenableFuture<Boolean> connect() {
+    return client.connect();
+  }
+
+  @Override
+  protected void setConnectionStatus(Throwable t) {
+
+  }
+
+  @Override
+  public void setConnectionTimeout(int timeout) {
+    client.setConnectionTimeout(timeout);
+  }
+
+  @Override
+  public int getTimeout() {
+    return client.getTimeout();
+  }
+
+  @Override
+  public int getWriteBufferSize() {
+    return client.getWriteBufferSize();
+  }
+
+  @Override
+  protected ByteBuffer getWriteBuffer() {
+    return null;
```

Comment:
I mean it should not really matter, this is only called by the socketExecuter and this client can never technically be put into one.  It just wraps a TCPClient to give the same onRead/onClose behavor, but its only ever the TCPClient that is used by the socketExecuter.  

I will go ahead and change this, but I am also not convinved extending Client is a good idea here.  I feel like it might be worth just setting up its own pattern instead of using the Client pattern.  Let me know what you think.